src/protocol/context/malicious.rs

```diff
@@ -24,36 +25,59 @@ use crate::secret_sharing::replicated::{
     malicious::AdditiveShare as MaliciousReplicated, semi_honest::AdditiveShare as Replicated,
 };
 use crate::secret_sharing::Arithmetic;
-use crate::sync::Arc;
 
 /// Represents protocol context in malicious setting, i.e. secure against one active adversary
 /// in 3 party MPC ring.
 #[derive(Clone, Debug)]
-pub struct MaliciousContext<'a, F: Field> {
-    /// TODO (alex): Arc is required here because of the `TestWorld` structure. Real world
-    /// may operate with raw references and be more efficient
-    inner: Arc<ContextInner<'a, F>>,
+pub struct MaliciousContext<'c, 'a, F: Field> {
+    inner: &'c ContextInner<'c, 'a, F>,
     step: Step,
     total_records: TotalRecords,
+    // Ensure that the struct is invariant over 'a, because that may be required in the future.
+    _dummy: PhantomData<OnceCell<&'a ()>>,
 }
 
-pub trait SpecialAccessToMaliciousContext<'a, F: Field> {
+/// Represents protocol context in malicious setting, i.e. secure against one active adversary
+/// in 3 party MPC ring.
+#[derive(Debug)]
+pub struct MaliciousContextBuf<'c, 'a, F: Field> {
```

Comment:
Would it be better to name this `MaliciousContext` and then rename `MaliciousContext` as `MaliciousContextRef` to reflect that the latter is really just a reference to the active state?

Reply:
I thought about that... I think one reason I didn't was to try and keep the diff smaller, but at that point I was still hoping not to change the lifetime parameter. If we have to touch all the instances of `SemiHonestContext<...>` to add the new lifetime parameter, then it's no more work to change the name. I don't have a strong preference between `Context` / `ContextBuf` and `ContextRef` / `Context`.

Reply:
I may also prefer `ContextRef`/`Context`. `Buf` to me implies that something may be growing or shrinking